sympy/plotting/tests/test_plot.py

```diff
@@ -623,3 +635,30 @@ def test_issue_13516():
     p = plot(sin(x), show=False)
     assert p.backend == DefaultBackend
     assert len(p[0].get_segments()) >= 30
+
+def test_check_figures_equal():
+    from sympy.testing.tmpfiles import TmpFileManager
+    matplotlib = import_module('matplotlib' , min_module_version='1.1.0', catch= (RuntimeError,))
+    if not matplotlib:
+        skip("Matplotlib not the default backend")
+    from matplotlib.testing.compare import compare_images
+    test_directory = os.path.dirname(os.path.abspath(__file__))
+
+    def tmp_file(dir=None, name=''):
+        return NamedTemporaryFile(suffix='.png', dir=dir, delete=False).name
+    try:
+        temp_dir = mkdtemp()
+        TmpFileManager.tmp_folder(temp_dir)
+        x = Symbol('x')
+        test_filename = tmp_file(dir = temp_dir, name = "test_singularity1")
+        cmp_filename = os.path.join(test_directory, "test_singularity1.png")
+        p = plot(x/(x-1)*(x-2))
+        p.save(test_filename)
+        compare_images(cmp_filename , test_filename , 0.005)
+        test_filename = tmp_file(dir = temp_dir, name = "test_singularity2")
+        cmp_filename = os.path.join(test_directory, "test_singularity2.png")
+        p = plot(sin(x)/x)
+        p.save(test_filename)
+        compare_images(cmp_filename , test_filename , 0.005)
+    finally:
+        TmpFileManager.cleanup()
```

Comment:
Is it not possible to use with?

Reply:
Yes, we can use with.
This test was added by taking #16674 as reference. But image comparison test is passing even if the file name is replaced with another file in the same directory.
So I planned to remove this test.